actors/builtin/miner/deadline_state.go

```diff
@@ -489,11 +497,12 @@ func (dl *Deadline) TerminateSectors(
 
 		// Record that partition now has pending early terminations.
 		dl.EarlyTerminations.Set(partIdx)
-		// Record live sector change
+		// Record change to sectors and power
 		dl.LiveSectors -= uint64(len(partSectors))
+		dl.FaultyPower = dl.FaultyPower.Sub(removed.FaultyPower)
 
-		// update power
-		removedPower = removedPower.Add(pwr)
```

Comment:
This previously counted all power from terminated sectors, not only the active power.